include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -1294,11 +1289,20 @@ __pattern_lexicographical_compare(_ExecutionPolicy&& __exec, _Iterator1 __first1
     using _Iterator1DifferenceType = typename ::std::iterator_traits<_Iterator1>::difference_type;
     using _ReduceValueType = int32_t;
 
-    auto __identity_init_fn = acc_handler_lexicographical_compare<_Compare, _ReduceValueType>{__comp};
-    auto __identity_reduce_fn = [](_ReduceValueType __a, _ReduceValueType __b) -> _ReduceValueType {
+    auto __reduce_fn = [](_ReduceValueType __a, _ReduceValueType __b) {
         bool __is_mismatched = __a != 0;
         return __a * __is_mismatched + __b * !__is_mismatched;
     };
+    auto __transform_fn = [__comp](auto __gidx, auto __acc1, auto __acc2) {
+        auto __s1_val = __acc1[__gidx];
```

Comment:
Is it better to store `__s1_val` and `__s2_val` as const references since we do not need a local copy of them?

Reply:
I changed that.